### ISSUE TITLE
PyCall functionality does not work

### ISSUE THREAD

**Issue by JakobAsslaender** [MagneticResonanceImaging/BartIO.jl#13]
@aTrotier : I am really struggling with getting the PyCall functionality working, on all 3 computers that I am using. My latest error that I am getting is:

```
julia> bart = wrapper_bart(expanduser("/gpfs/home/asslaj01/Documents/toolboxes/bart/"))
Requirement already satisfied: numpy in /gpfs/scratch/asslaj01/.julia/conda/3/lib/python3.9/site-packages (1.22.3)
ERROR: PyError (PyImport_ImportModule

The Python package bart could not be imported by pyimport. Usually this means
that you did not install bart in the Python version being used by PyCall.

PyCall is currently configured to use the Julia-specific Python distribution
installed by the Conda.jl package.  To install the bart module, you can
use `pyimport_conda("bart", PKG)`, where PKG is the Anaconda
package that contains the module bart, or alternatively you can use the
Conda package directly (via `using Conda` followed by `Conda.add` etcetera).

Alternatively, if you want to use a different Python distribution on your
system, such as a system-wide Python (as opposed to the Julia-specific Python),
you can re-configure PyCall with that Python.   As explained in the PyCall
documentation, set ENV["PYTHON"] to the path/name of the python executable
you want to use, run Pkg.build("PyCall"), and re-launch Julia.

) <class 'ModuleNotFoundError'>
ModuleNotFoundError("No module named 'bart'")

Stacktrace:
 [1] pyimport(name::String)
   @ PyCall ~/.julia/packages/PyCall/ygXW2/src/PyCall.jl:558
 [2] wrapper_bart(pathtobart::String)
   @ BartIO ~/.julia/dev/BartIO/src/BartIO.jl:43
 [3] top-level scope
   @ REPL[6]:1
```

And, worst of all, these problems prevents me from using the read/write functionality and hence, makes the entire package un-usable. So my question is, 
1. what are you using Pycall.jl for? Would it also be possible to just use system calls to BART?
2. would it be worth to re-structure the toolbox -- even though it is already small so that BartIO maybe just contains the read/write functionality and a separate package Bart.jl contains the wrapper functionality? Or are there other ways to ensure that the read/write functionality remains usable, even if PyCall does not precompile or is otherwise unusable?

**Comment by aTrotier** [comment]
Hi,
Weird, it really works without any issue on both my mac and on a ubuntu server (and on the CI). What version of bart are you trying to call ?

1) Pycall is used only to call the system calls. sure we can create one direct wrapper that just write a .cfl in a tmp folder, use the system call, read the cfl and delete them. But we have to manage the input/ouput and passing the parameters values correctly and keeping the version uptodate.

2) I don't understand why you can't use `read_cfl`and `write_cfl`. It is a pure julia function and does not use PyCall. after writing you can use the system call to perform the reconstruction. I am actually using that function to wrote basis / density compensation function for subspace/non-cartesian reconstruction and then use the wrapper.

**Comment by JakobAsslaender** [comment]
I am trying to use the latest version and on my M1 Mac laptop, the precompile fails entirely. 

Why are we not using [Julia's system calls](https://docs.julialang.org/en/v1/manual/running-external-programs/)?

**Comment by aTrotier** [comment]
We can, we just need to convert that in julia : https://github.com/mrirecon/bart/blob/master/python/bart.py or the matlab one.

The precompilation fails only on your mac M1 ?

**Comment by JakobAsslaender** [comment]
Hmm, I think removing the third language from the mix would make this package more robust, wouldn't it?

Yes, it seems on my Intel Mac and the Linux (also x86) the precompile works, but the test and calls of the wrapper fail.

**Comment by aTrotier** [comment]
Sure, it was just a question of development time. Now, BartIO is used in multiple projects, it makes sense to remove the python call. I'll work on that

**Comment by aTrotier** [comment]
Implemented by #14
